containerize/containerize.sh

```diff
@@ -0,0 +1,195 @@
+
+BASE=
+KEEPCACHE="--exclude-caches-all"
+CLEAN="false"
+DIR="/home/cmsusr"
+FILE="Dockerfile"
+JSON="${SCRIPTPATH}/cache.json"
+TAG="analysis"
+TAR=""
+VCS="--exclude-vcs"
+XRDIR=""
+
+usage(){
+    EXIT=$1
+
+    echo "containerize.sh [options]"
+    echo ""
+	echo "-b [base]           the base image to use (default = ${BASE})"
+	echo "-c                  keep the cached files when making the tarball (default = False)"
+	echo "-C                  cleanup the temporary files when finished making the image (default = ${ClEAN})"
+    echo "-d [dir]            project installation area inside the container (default = ${DIR})"
+    echo "-f [file]           the Dockerfile to use to build the image (default = ${FILE})"
+    echo "-j [json]           path to the json file containing the path to cache (default = ${JSON})"
+    echo "-t [tag]            the tag to use for the image (default = ${TAG})"
+	echo "-T [tar]            path to an existing tarball to use (default = ${TAR})"
+	echo "-v                  include the vcs directories (default = False)"
+	echo "-x [XRDIR]          the path to the tarball on an XRootD accessible device (default = ${XRDIR})"
+    echo "-h                  display this message and exit"
+	echo
+	echo "Examples:"
+	echo "./containerize.sh -t <tag> -b docker://docker.io/cmscloud/cc7-cms"
+	echo "podman run --rm -it -v /cvmfs/cms.cern.ch/:/cvmfs/cms.cern.ch/:ro 4a73e8eb3bf5f2a3ab84e2eedfb314e3489529e2b02a464cd7eeb0d2782091ff"
+	echo "./containerize.sh -t <tag> -b docker://docker.io/aperloff/cms-cvmfs-docker:light -C -v"
+	echo "podman run --rm -it <tag>"
+
+    exit ${EXIT}
+}
+
+# process options
+while getopts "b:cCd:f:j:t:T:vx:h" opt; do
+    case "$opt" in
+	b) BASE=$OPTARG
+	;;
+    c) KEEPCACHE=""
+    ;;
+	C) CLEAN="true"
+    ;;
+    d) DIR=$OPTARG
+    ;;
+    f) FILE=$OPTARG
+    ;;
+    j) JSON=$OPTARG
+    ;;
+    t) TAG=$OPTARG
+    ;;
+    T) TAR=$OPTARG
+    ;;
+	v) VCS=""
+    ;;
+    x) XRDIR=$OPTARG
+    ;;
+    h) usage 0
+    ;;
+    esac
+done
+
+dependency_check() {
+	subuid=$(cat /etc/subuid | grep "^`id -u`:")
+	subgid=$(cat /etc/subgid | grep "^`id -u`:")
+	if [ ! command -v buildah &> /dev/null ]; then
+		EXIT=$?
+		echo "Buildah could not be found!"
+		exit ${EXIT}
+	elif [[ -z "${subuid}" ]] || [[ -z "${subgid}" ]]; then
+		echo "Unable to find a subuid or subgid for id=`id -u` in /etc/subuid and /etc/subgid."
+		echo "Contact user support or your sysadmin for further assistance."
+	elif [[ ! -d /cvmfs/cms.cern.ch ]] || [ ! cvmfs_config probe cms.cern.ch &> /dev/null ]; then
+		EXIT=$?
+		echo "/cvmfs/cms.cern.ch must be mounted on the host to proceed."
+		exit ${EXIT}
+	fi
+
+	if [ ! command -v podman &> /dev/null ]; then
+		echo "Podman could not be found. While this is not strictly necessary, you will not be able to create a container from the resulting image."
+	fi
+}
+dependency_check
+
+# cache unneeded files
+CACHEDIR="""\
```

Comment:
I don't think triple quotes do anything special in bash, so this could just be wrapped in single quotes

Reply:
Changed to single quote.